### PR TITLE
DOCS: add CloudEvent parsing support

### PR DESCRIPTION
**Pull request by stijnmoreels**
Relates to #8

### CODE REVIEW

File: docs/features/deserializing-events.md [hunk 1]
```diff
@@ -30,4 +34,43 @@ When using official Azure events, you can use `.ParseFromData<>` to deserialize
 EventGridBatch<EventGridEvent<StorageBlobCreatedEventData>> eventGridBatch = EventGridParser.ParseFromData<StorageBlobCreatedEventData>(rawEvent);
 ```
 
+### Deserializing Both Cloud and EventGrid Events
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]: I would move this before `Deserializing Custom Events`

File: docs/features/deserializing-events.md [hunk 2]
```diff
@@ -30,4 +34,43 @@ When using official Azure events, you can use `.ParseFromData<>` to deserialize
...
+
+We provide support for deserializing Cloud Events using the [official SDK for C#](https://github.com/cloudevents/sdk-csharp).
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]:
```suggestion
We provide support for deserializing CloudEvents using the [official SDK for C#](https://github.com/cloudevents/sdk-csharp).
```

File: docs/features/deserializing-events.md [hunk 3]
```diff
@@ -30,4 +34,43 @@ When using official Azure events, you can use `.ParseFromData<>` to deserialize
...
+
+Upon receiving of EventGrid events and Cloud events, you can use the same deserialization functionality.
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]:
```suggestion
Upon receiving of EventGrid events and CloudEvents, you can use the same deserialization functionality.
```

File: docs/features/deserializing-events.md [hunk 4]
```diff
@@ -30,4 +34,43 @@ When using official Azure events, you can use `.ParseFromData<>` to deserialize
...
+
+Deserializing Cloud events:
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]:
```suggestion
Deserializing CloudEvents:
```

File: docs/features/deserializing-events.md [hunk 5]
```diff
@@ -21,6 +23,8 @@ CarEventData typedEventPayload = eventGridMessage.GetPayload();
 object untypedEventPaylaod = eventGridMessage.Data;
 ```
 
+### Deserializing Azure Events
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]: I would emphasize that this is for built-in Azure events
```suggestion
### Deserializing built-in Azure Events
```

File: docs/features/deserializing-events.md (hunk 1 above)
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]: Now it's not really clear what the section is about, how about this?
```suggestion
### Deserializing Events with CloudEvent or EventGrid schema
```

Comment (tomkerkhove) [reply docs/features/deserializing-events.md]: I would actually create 2 sections, one per schema.

File: docs/features/deserializing-events.md [hunk 6]
```diff
@@ -5,6 +5,8 @@ layout: default
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]: I would provide a brief introduction on what this doc is and the supported ways to deserialize events with a pointer to the doc section (list-approach for example)

File: docs/features/deserializing-events.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/features/deserializing-events.md]: Done.

File: docs/features/deserializing-events.md (hunk 6 above)
Comment (stijnmoreels) [reply docs/features/deserializing-events.md]: Added one.

File: docs/features/deserializing-events.md [hunk 7]
```diff
@@ -5,6 +5,57 @@ layout: default
 
 ## Deserializing Events
 
+The `Arcus.EventGrid` package provides several ways to deserializing events.
+Following paragraphs describe each supported type of event.
+
+- [Deserializing EventGrid Events](#deserializing-eventgrid-events)
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]: This is the official name while `CloudEvents` is the official name for that spec
```suggestion
- [Deserializing Event Grid Events](#deserializing-event-grid-events)
```

File: docs/features/deserializing-events.md [hunk 8]
```diff
@@ -5,6 +5,57 @@ layout: default
...
+- [Deserializing CloudEvent Events](#deserializing-cloudevent-events)
+- [Deserializing Custom Events](#deserializing-custom-events)
+- [Deserializing Built-In Azure Events](#deserializing-built-in-azure-events)
+
+
+### Deserializing EventGrid Events
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]:
```suggestion
### Deserializing Event Grid Events
```

File: docs/features/deserializing-events.md [hunk 9]
```diff
@@ -5,6 +5,57 @@ layout: default
 
 ## Deserializing Events
 
+The `Arcus.EventGrid` package provides several ways to deserializing events.
+Following paragraphs describe each supported type of event.
+
+- [Deserializing EventGrid Events](#deserializing-eventgrid-events)
+- [Deserializing CloudEvent Events](#deserializing-cloudevent-events)
+- [Deserializing Custom Events](#deserializing-custom-events)
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]: Let's move this to the bottom of the list

File: docs/features/deserializing-events.md [hunk 10]
```diff
@@ -5,6 +5,57 @@ layout: default
...
+- [Deserializing Built-In Azure Events](#deserializing-built-in-azure-events)
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]: Let's move this to first item

File: docs/features/deserializing-events.md [hunk 11]
```diff
@@ -5,6 +5,57 @@ layout: default
 
 ## Deserializing Events
 
+The `Arcus.EventGrid` package provides several ways to deserializing events.
+Following paragraphs describe each supported type of event.
+
+- [Deserializing EventGrid Events](#deserializing-eventgrid-events)
+- [Deserializing CloudEvent Events](#deserializing-cloudevent-events)
```
Comment (tomkerkhove) [inline docs/features/deserializing-events.md]: Let's move this to second as we should aim to use CloudEvents over Event Grid native events

File: docs/index.md [hunk 12]
```diff
@@ -17,7 +17,7 @@ PM > Install-Package Arcus.EventGrid.All
 - Support for publishing events to an Event Grid Topic ([docs](features/publishing-events))
 - Support for handling endpoint validation handshake ([docs](features/endpoint-validation))
 - Provides tooling for writing integration tests ([docs](features/running-integration-tests))
-- Support for deserializing custom & official Azure events ([docs](features/deserializing-events))
+- Support for deserializing EventGrid, CloudEvent, custom & official Azure events ([docs](features/deserializing-events))
```
Comment (tomkerkhove) [inline docs/index.md]: This is more clear I think - First 2 are about the schema of data while the latter about the envelope
```suggestion
- Support for deserializing official Azure or custom events according to Event Grid & CloudEvent schema ([docs](features/deserializing-events))
```

**Pull request merged**